bootstrap/issuers/helmfile/charts/letsencrypt/templates/letsencrypt.yaml

```diff
@@ -1,6 +1,6 @@
 {{ if .Values.letsencrypt.enabled -}}
 ---
-apiVersion: cert-manager.io/v1alpha2
+apiVersion: cert-manager.io/v1
```

Comment:
If we do this change in conjucntion with upgrading cert-manager we will have to have a somewhat complicated upgrade path if one uses letsencrypt issuers.
This is because of our workflow:
Bootstrap: upgrade crds -> upgrade letsencrypt issuers which will fail as we have only upgraded the certmanager crds but not the application!
We will get the following error:
```
Error: UPGRADE FAILED: rendered manifests contain a resource that already exists. Unable to continue with update: could not get information about the resource: conversion webhook for cert-manager.io/v1alpha2, Kind=Issuer failed: the server rejected our request for an unknown reason
```

This means that our upgrade path would become
```
# Upgrade crds, wait for it to fail
./bin/ck8s bootstrap {wc.sc}

./bin/ck8s ops helmfile {wc,sc} -l app=cert-manager -i apply

./bin/ck8s bootstrap {wc.sc}
```

Reply:
Maybe I'm just a bit confused,i need to investigate further

Reply:
I think you are correct that cert-manager must be upgraded before the issuers. So if we want all in one go, it should be
1. Upgrade CRDs
2. Upgrade cert-manager
3. Upgrade issuers/certificates and other resources

I'm fine with leaving the issuers/certificates on v1alpha2 for the next release to keep the migration simple.

Reply:
The thing is that is we leave the issuers at v1alpha2 we will not have any problem when upgrading but problems when installing on a fresh cluster as the it will try to call the cert-manager conversion webhook which has not yet been installed.
So, it's a bit of a chicken and egg problem in either case :smile:

Reply:
Anyways, I tried with the current solution a fresh install in the workload cluster and an upgrade in the serice cluster by upgrading cert-manager before the crds and it seems to work
```
./bin/ck8s ops helmfile sc -l app=cert-manager -i apply
./bin/ck8s apply sc
```

Reply:
As we decided to move the issuers from the bootstrap step to apps this issue is not longer relevant :)